avocado/plugins/run.py

```diff
@@ -77,7 +77,10 @@ def configure(self, parser):
                                   'You can also use suffixes, like: '
                                   ' s (seconds), m (minutes), h (hours). '))
 
-        parser.add_argument('--failfast', action='store_true',
+        parser.add_argument('--failfast',
+                            nargs='?',
+                            const=True,
+                            type=self._failfast,
                             help='Interrupt job on first failed test.')
```

Comment:
I think you should improve this help as well as it could be a bit misleading (argparse asks for optional argument, but they are not listed there, nor do we know what happens without them).